### ISSUE TITLE
Trying to hook a system daemon and write file, but get Operation not permitted

### ISSUE THREAD

**Issue by liuxuan30** [pwn20wndstuff/Undecimus#1927]
**Describe the bug**
Trying to hook a system daemon and write a temp file to `/Library/MobileSubstrate/DynamicLibraries`, but get Operation not permitted

**To Reproduce**
1. hook adprivacyd and try to write a file like
```objc
        NSError *fileErr;
        NSString *str = @"hello world";
        BOOL written = [str writeToFile:@"/Library/MobileSubstrate/DynamicLibraries/helloworld.txt" atomically:YES encoding:4 error:&fileErr];
        DDLog(@"write ok? %d error:%@", written, fileErr);
```
I tried other directory path, still no luck.

FYI: I originally want to start a web server in this daemon, while trying to start server, it also throws this same error. I suspect it has something to do with the access control, but don't know whether it's jailbreak issue or, it's the daemon's issue. Thanks!

**Device (please complete the following information):**
 - iOS Version: 12.0.1
 - iOS Device: iphone 7
 - unc0ver Version: 4.3.1

**Place an "x" between the brackets if true:**
 - [x] this is a bug others will be able to reproduce
 - [x] this issue is present with all tweaks uninstalled(except for default packages) or disabled
 - [x] this issue is present after a rootfs restore
 - [x] this issue is present on the latest version of unc0ver

**Comment by Cryptiiiic** [comment]
Learn proper tweak/process/sandbox etiquette. Obviously you can't write to a root directory from a sandboxed mobile process such as adprivacyd. And obviously the only thing that should ever be writing to /Library/MobileSubstrate or /Library/MobileSubstrate/DynamicLibraries/* is dpkg itself. Learn better practices please. You need to write to an actual sandbox mobile temp directory from a sandbox mobile process. Examples being /tmp, /var/log, /var/logs, /var/mobile/Library or /var/mobile/Library/*
Don't waste space creating absolutely useless unc0ver issues if you don't even know the fundamental basics of which you are basing the issue on. @liuxuan30

**Comment by liuxuan30** [comment]
Hi @Cryptiiiic thanks for the reminding. I indeed to visit those again. I'm writing because I found I cannot write into none of your above paths in a system daemon called adprivayd, but was able to write in other daemons like appstored. I'm kind of confused because I didn't see what entitlements are related to such restriction, so I asked here. 

update: it seems related to a seatbelt profile:
```
	<key>seatbelt-profiles</key>
	<array>
		<string>adtrackingd</string>
	</array>
```
